docs/user-guide/02-convenience-features.md

```diff
@@ -0,0 +1,227 @@
+
+```hlsl
+struct Foo
+{
+    int count;
+    
+    [mutating]
+    void setCount(int x) { count = x; }
+
+    void setCount2(int x) { count = x; }
+}
+
+void test()
+{
+    Foo f;
+    f.setCount(1); // f.count is 1 after the call.
+    f.setCount2(2); // f.count is still 1 after the call.
+}
+```
+
+## Properties
+
+Properties provide a convienient way to access values exposed by a type, where the logic behind accessing the value is defined in `getter` and `setter` function pairs. Slang's `property` feature is similar to C# and Swift. 
+```C#
+struct MyType
+{
+    uint flag;
+
+    property uint highBits
+    {
+        get { return flag >> 16; }
+        set { flag = (flag & 0xFF) + (newValue << 16); }
+    }
+};
+```
+
+Or equivalently in a "modern" syntax:
+
+```C#
+struct MyType
+{
+    uint flag;
+
+    property highBits : uint
+    {
+        get { return flag >> 16; }
+        set { flag = (flag & 0xFF) + (newValue << 16); }
+    }
+};
+```
+
+You may also use an explicit parameter for the setter method:
+```C#
+property uint highBits
+{
+    set(uint x) { flag = (flag & 0xFF) + (x << 16);  }
+}
+```
+
+> #### Note ####
+> Slang currently does not support automatically synthesized `getter` and `setter` methods. For example,
+> the following code is not supported:
+> ```
+> property uint highBits {get;set;} // Not supported yet.
+> ```
+
+## Initializers
+> #### Note ####
+> The syntax for defining initializers are subject to future change.
+
+
+Slang supports defining initializers in `struct` types. You can write:
+```C#
+struct MyType
+{
+    int myVal;
+    __init(int a, int b)
+    {
+        myVal = a + b;
+    }
+}
+```
+
+You can use an initializer to construct a new instance by using the type name in a function call expression:
+```C#
+MyType instance = MyType(1,2);  // instance.myVal is 3.
+```
+
+You may also use C++ style initializer list to invoke an initializer:
+```C#
+MyType instance = {1, 2};
+```
+
+If an initializer does not define any parameters, it will be recognized as *default* initializer that will be automatically called at the definition of a variable:
+
+```C#
+struct MyType
+{
+    int myVal;
+    __init(i)
+    {
+        myVal = 10;
+    }
+};
+
+int test()
+{
+    MyType test;
+    return test.myVal; // returns 10.
+}
+```
+
+## Operator Overloading
+
+Slang allows defining operator overloads as global methods:
+```C#
+struct MyType
+{
+    int val;
+    __init(int x) { val = x; }
+}
+
+MyType operator+(MyType a, MyType b)
+{
+    return MyType(a.val + b.val);
+}
+
+int test()
+{
+    MyType rs = MyType(1) + MyType(2);
+    return rs.val; // returns 3.
+}
+```
+
+Slang currently supports overloading the following operators: `+`, `-`, `*`, `/`, `%`, `&`, `&&`, `|`, `||`, `<`, `>`, `<=`, `>=`, `==`, `!=`, unary `-`, `~` and `!`.
```

Comment:
This list is missing the compound assignment operators like `+=`.

Realistically, we probably want to have a POR of defining some built-in `interface`s for the typical interpretation of the binary/unary operators, and then allow types to opt in to operator support by conforming to those interfaces (that is how Rust works). This approach minimizes the total number of operator overloads that the semantic checker needs to deal with, and also means that opting into the `IAddable` interface (or whatever it is called) automatically confers both support for `+` and `+=`. A similar statement could be made for an `IPartiallyOrdered` and `ITotallyOrdered` protocol pair, which could be used to implement all the comparison ops (giving many of the same benefits that recent C++ adds with the "spaceship operator").